sdk/core/Azure.Core/tests/FailedResponseExceptionTests.cs

```diff
@@ -183,6 +183,50 @@ public void RequestFailedExceptionIsSerializeable()
             Assert.AreEqual(exception.ErrorCode, deserialized.ErrorCode);
         }
 
+        [Test]
+        public async Task ParsesJsonErrors()
+        {
+            var formattedResponse =
+                "Custom message" + s_nl +
+                "Status: 210 (Reason)" + s_nl +
+                "ErrorCode: StatusCode" + s_nl +
+                s_nl +
+                "Content:" + s_nl +
+                "{ \"error\": { \"code\":\"StatusCode\", \"message\":\"Custom message\" }}" + s_nl +
+                s_nl +
+                "Headers:" + s_nl +
+                "Content-Type: text/json" + s_nl;
+
+            var response = new MockResponse(210, "Reason");
+            response.SetContent("{ \"error\": { \"code\":\"StatusCode\", \"message\":\"Custom message\" }}");
+            response.AddHeader(new HttpHeader("Content-Type", "text/json"));
+
+            RequestFailedException exception = await ClientDiagnostics.CreateRequestFailedExceptionAsync(response);
+            Assert.AreEqual(formattedResponse, exception.Message);
+            Assert.AreEqual("StatusCode", exception.ErrorCode);
+        }
+
+        [Test]
+        public async Task IgnoresInvalidJsonErrors()
```

Comment:
nit: how about a test for valid json that doesn't have the expected shape?

Reply:
Done.